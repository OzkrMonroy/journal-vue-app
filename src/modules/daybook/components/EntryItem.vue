<template>
  <div
    class="entry-container mb-3 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: entry.id } })"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">{{ entryDay }}</span>
      <span class="mx-1 fs-5">{{ entryMonth }}</span>
      <span class="mx-2 fw-light">{{ entryYear }}</span>
    </div>
    <div class="entry-description">
      <p>
        {{ shortText }}
      </p>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    entry: {
      type: Object,
      required: true,
    },
  },
  computed: {
    shortText() {
      return this.entry.text.length > 100
        ? this.entry.text.substr(0, 100) + "..."
        : this.entry.text;
    },
    entryDay() {
      const day = new Date(this.entry.date).getDate();
      return day;
    },
    entryMonth() {
      const moth = new Date(this.entry.date).toLocaleString("default", {
        month: "long",
      });
      return moth;
    },
    entryYear() {
      const year = new Date(this.entry.date).toLocaleString("default", {
        year: "numeric",
        weekday: "long",
      });
      return year;
    },
  },
};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }
}
.entry-description {
  font-size: 12px;
  text-align: left;
}
</style>
