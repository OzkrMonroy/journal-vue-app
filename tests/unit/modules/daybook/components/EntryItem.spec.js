import { shallowMount } from "@vue/test-utils";
import EntryItem from "@/modules/daybook/components/EntryItem.vue";
import { journalState } from "../../../mocks/test-journal-state";

describe("Tests for EntryItem component", () => {
  const mockRouter = {
    push: jest.fn(),
  };
  const entry = journalState.entries[0];
  const wrapper = shallowMount(EntryItem, {
    props: {
      entry,
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test("should render the component correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should redirect when the user click in the entry-container", () => {
    wrapper.find(".entry-container").trigger("click");

    expect(mockRouter.push).toHaveBeenCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: entry.id },
    });
  });

  test("should return the date data correctly", () => {
    expect(wrapper.vm.entryDay.toString()).toBe("6");
    expect(wrapper.vm.entryMonth).toBe("agosto");
    expect(wrapper.vm.entryYear).toBe("2022 s√°bado");
  });
});
