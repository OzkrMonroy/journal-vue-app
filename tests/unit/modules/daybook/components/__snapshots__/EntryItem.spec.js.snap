// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for EntryItem component should render the component correctly 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">6</span><span class="mx-1 fs-5">agosto</span><span class="mx-2 fw-light">2022 s√°bado</span></div>
  <div class="entry-description">
    <p>I'm learning vue again!</p>
  </div>
</div>
`;
